linker-interface/shared/src/main/scala/org/scalajs/linker/interface/ModuleOutputSpec.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker.interface
+
+/** Module output specification. */
+final class ModuleOutputSpec private (modules: Seq[(ModuleOutputSpec.Selector, ModuleOutputSpec.Module)]) {
+  private def this() = this(Nil)
+
+  def addModule(selector: ModuleOutputSpec.Selector, module: ModuleOutputSpec.Module): ModuleOutputSpec = {
+    // TODO ensure selectors don't conflict.
+    ???
+    new ModuleOutputSpec(modules :+ (selector, module))
+  }
+}
+
+object ModuleOutputSpec {
+  def apply(): ModuleOutputSpec = new ModuleOutputSpec()
+
+  final class ModuleID private[ModuleOutputSpec] (val id: String) extends AnyVal
+
+  final class Module private (
+      val id: ModuleID,
+      initializers: Seq[ModuleInitializer],
+      deps: Seq[ModuleID]
+  ) {
+    private def this(id: ModuleID) = this(id, Nil, Nil)
+
+    def withInitializers(initializers: ModuleInitializer*): Module =
+      copy(initializers = initializers)
+
+    def withDeps(deps: ModuleID*): Module =
```

Comment:
The idea here is that in a future version we can add `withAutoDeps` if this is a feature that is needed.

Reply:
If dependencies must/can be explicitly specified, what happens if they contradict the reality? For example, if I forget to declare a dependency from A to B, but in practice a class in A references a class in B?

If we have to detect those issues, why can't we *always* detect dependencies, making this explicitly configuration redundant?

Reply:
> How does the caller of withDeps create ModuleIDs? It seems there is no public way to construct one.

`Module#apply` creates one from a string.

> If dependencies must/can be explicitly specified, what happens if they contradict the reality?

Does this answer your question: https://github.com/scala-js/scala-js/issues/2681#issuecomment-581153171?

tl;dr IMHO it is important to make sure there are no *unwanted* dependencies. Maybe we should call this `withAllowedStaticDependencies` or something like that.